### PR TITLE
💚 Trigger CLI Regression on PR comment

### PR DESCRIPTION
**Pull request by samarsault**
Runs regression tests for all SDKs when "RUN_REGRESSIONS" is commented

### CODE REVIEW

File: .github/workflows/sdk-regression.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+on:
+  pull_request:
+    types: [opened]
+  issue_comment:
+    types: [created]
```
Comment (itsjwala) [inline .github/workflows/sdk-regression.yml:5]:
```suggestion
    types: [created, edited]
```

* lets allow edited comment as well
  *  say we edited the comment later to correct name post some typo -> it should trigger this workflow.

File: .github/workflows/sdk-regression.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Trigger Workflow
+        uses: actions/github-script@v6
+        with:
+          script: |
+            github.rest.actions.createWorkflowDispatch({
+              owner: 'percy',
+              repo: ${{ matrix.repo}}
+              workflow_id: 'test.yml',
+              ref: '${{ github.ref }}',
+            })
```
Comment (itsjwala) [inline .github/workflows/sdk-regression.yml:38]: EOF

File: .github/workflows/sdk-regression.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      run-regression: ${{steps.regcheck.outputs.triggered}}
+    steps:
+      - uses: khan/pull-request-comment-trigger@master
+        name: Check if regression
+        if: github.event_name == 'pull_request'
+        id: regcheck
+        with:
+          trigger: 'RUN_SDK_REGRESSION'
```
Comment (itsjwala) [inline .github/workflows/sdk-regression.yml:19]:
```suggestion
          trigger: 'RUN_REGRESSION'
```

File: .github/workflows/sdk-regression.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+on:
+  pull_request:
+    types: [opened]
+  issue_comment:
+    types: [created]
+
+jobs:
+  check-regression:
+    name: Check if needs to run sdk regression
```
Comment (itsjwala) [inline .github/workflows/sdk-regression.yml:9]:
```suggestion
    name: Check if needs to run sdk regression
    if: ${{ github.event.issue.pull_request }}
```

run only on issue type = PR, [ref](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment-on-issues-only-or-pull-requests-only)

File: .github/workflows/sdk-regression.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+on:
+  pull_request:
+    types: [opened]
+  issue_comment:
+    types: [created]
+
+jobs:
+  check-regression:
```
Comment (itsjwala) [inline .github/workflows/sdk-regression.yml:8]:
```suggestion
  regression:
```

File: .github/workflows/sdk-regression.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    outputs:
+      run-regression: ${{steps.regcheck.outputs.triggered}}
+    steps:
+      - uses: khan/pull-request-comment-trigger@master
+        name: Check if regression
+        if: github.event_name == 'pull_request'
+        id: regcheck
+        with:
+          trigger: 'RUN_SDK_REGRESSION'
+  sdk-regression:
```
Comment (itsjwala) [inline .github/workflows/sdk-regression.yml:20]: do we need a separate `job` for this, can this be indented under `regression:` job - easier to navigate and debug?

File: .github/workflows/sdk-regression.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    types: [created]
+
+jobs:
+  check-regression:
+    name: Check if needs to run sdk regression
+    runs-on: ubuntu-latest
+    outputs:
+      run-regression: ${{steps.regcheck.outputs.triggered}}
+    steps:
+      - uses: khan/pull-request-comment-trigger@master
```
Comment (itsjwala) [inline .github/workflows/sdk-regression.yml:14]: this dependency can be removed with something like

```yml
if: ${{ github.event.comment.body == 'RUN_REGRESSION'}}
```

File: .github/workflows/sdk-regression.yml (hunk 6 above)
Comment (samarsault) [reply .github/workflows/sdk-regression.yml:15]: Won't it be easier to navigate debug if we have a separate job? We'll be able to see when it triggered and when it did not without getting into the logs

**Comment by samarsault** [comment]
RUN_REGRESSION

**Review by itsjwala — APPROVED** [review]

**Pull request merged**